BMSUtils.cpp

```diff
@@ -0,0 +1,108 @@
+#include <iostream>
+#include <assert.h>
+#include "BMSUtils.h"
+
+
+Parameter::Parameter(Language Lang, float min_value, float max_value, string name)
+{
+    min_threshold = min_value;
+    max_threshold = max_value;
+    parameter_name = name;
+    tolerance = max_threshold * 0.05 ;
+    if(Lang == ENGLISH){
+
+        error_msg = "Out of range";
+        warning_msg = "Close to warning range";
+
+    }
+    else if(Lang == DEUTSCH){
+        error_msg = "Außer Reichweite";
+        warning_msg = "In der Nähe des Warnbereichs";
+    }
+}
+
+bool Parameter :: check_min(float value){
+    if(value>=min_threshold)
+    {
+        if(value <= min_threshold + tolerance)
+            cout << parameter_name << " - "<< warning_msg <<endl;
+        return true;
+    }
+    else
+    {
+        cout << parameter_name << " - "<< error_msg <<endl;
+        return false;
+             
+    }
+       
+ 
+};
+
+bool Parameter ::  check_max(float value){
+
+    if(value<=max_threshold)
+        {
+            if(value >= max_threshold - tolerance)
+                cout << parameter_name << " - "<< warning_msg <<endl;
```

Comment:
Any way to accumulate the results and show them together?

Reply:
Sorry, i was not able to clearly understand this part as i was not able to attend last session, do i have to log the report in some file?